src/main/daml/Daml/Finance/Interface/Instrument/Swap/Fpml/FpmlTypes.daml

```diff
@@ -56,9 +56,31 @@ data PaymentDates = PaymentDates
     firstPaymentDate : Optional Date
     lastRegularPaymentDate : Optional Date
     payRelativeTo : DateRelativeToEnum
+    paymentDaysOffset : Optional DateOffset
     paymentDatesAdjustments : BusinessDayAdjustments
   deriving (Eq, Show)
 
+-- | A type defining an offset used in calculating a date when this date is defined in reference to
+-- another date through a date offset. The type includes the convention for adjusting the date and
+-- an optional sequence element to indicate the order in a sequence of multiple date offsets.
+data DateOffset = DateOffset with
```

Comment:
maybe indent as follows:

```
data Rounding = Rounding 
  with
    roundingDirection : RoundingDirectionEnum
      -- ^ Specifies the rounding direction.
    precision : Int
      -- ^ Specifies the rounding precision in terms of a number  of decimal places. Note how a
      -- percentage rate rounding of 5 decimal places is expressed as a rounding precision of 7 in
      -- the FpML document since the percentage is expressed as a decimal, e.g. 9.876543% (or
      -- 0.09876543) being rounded to the nearest 5 decimal places is 9.87654% (or 0.0987654).
  deriving (Eq, Show)
```